source/_integrations/tplink.markdown

```diff
@@ -35,50 +38,87 @@ There is currently support for the following device types within Home Assistant:
+|------------------|---------------------------|--------------------------------------------|------------------------------------------|
+| EP40             | ✅                        | ❌                                         | outdoor 2-outlet                         |
+| HS107            | ✅                        | ❌                                         | indoor 2-outlet                          |
+| HS300            | ✅                        | ❌                                         | 6-outlet<br>supports consumption sensors |
+| KP40             | ✅                        | ❓ <!-- not mentioned as of 2022-05-01 --> | outdoor 2-outlet                         |
+| KP200            | ✅                        | ❌                                         | indoor 2-outlet                          |
+| KP303            | ✅                        | ❌                                         | 3-outlet                                 |
+| KP400            | ✅                        | ❌                                         | outdoor 2-outlet                         |
 
 ### Wall Switches
 
-- HS200
-- HS210
-- HS220 (acts as a light)
+| Model Number     | Support by Home Assistant | Support by TP-Link's 'local only mode'     | Note                                     |
+|------------------|---------------------------|--------------------------------------------|------------------------------------------|
+| ES10             | ❓                        | ✅                                         |                                          |
+| HS200            | ✅                        | ✅                                         |                                          |
+| HS210            | ✅                        | ❌                                         |                                          |
+| HS210KIT         | ❓                        | ❌                                         |                                          |
+| HS220            | ✅                        | ❌                                         | acts as a light                          |
 
 ### Bulbs
 
-- LB100
-- LB110
-- LB120
-- LB130
-- LB230
-- KL110
-- KL120
-- KL125
-- KL130
-- KB130
+| Model Number     | Support by Home Assistant | Support by TP-Link's 'local only mode'     | Note                                     |
+|------------------|---------------------------|--------------------------------------------|------------------------------------------|
+| KB130            | ✅                        | ❓ <!-- not mentioned as of 2022-05-01 --> |                                          |
+| KL50             | ❓                        | ✅                                         |                                          |
+| KL60             | ❓                        | ✅                                         |                                          |
```

Comment:
These work. There are test fixtures for them here

https://github.com/python-kasa/python-kasa/tree/master/kasa/tests/fixtures